### ISSUE TITLE
Explain in detail the logic of cell reuse in `EpoxyCollectionView`?

### ISSUE THREAD

**Issue by rakuyoMo** [airbnb/epoxy-ios#110]
`EpoxyCollectionView` works very well and I've tried to build many pages with it, but I'm still confused about one thing: what does the reuse mechanism in Epoxy actually look like? When I provide what kind of data, it will update the list?

I know that dequeue is triggered when `dataID` and `Style` change, but I've been confused about `Content` related content.

Suppose there is a scenario: a list (`EpoxyCollectionView`) with a lot of data, with a `UITextField` on each `Cell`, which needs to listen to user input, while considering the reuse logic.

I got confused in this situation. I listened for user input inside `behavior`, then I tried to set the content for `UITextField` in `content`, but it had some problems when reuse occurred in `cell`, then I changed to use `setBehaviors` to assign a value to `textField.text` instead of the previous ` content`, but it still has problems in some "wrong usage".

I'll provide a demo later if needed to show my "misuse".

**Comment by rakuyoMo** [comment]
I will provide a demo at a later time

**Comment by thedrick** [comment]
@rakuyoMo does the TextFieldViewController Example help solve your question? https://github.com/airbnb/epoxy-ios/blob/master/Example/EpoxyExample/ViewControllers/CollectionView/TextFieldViewController.swift#L35-L47

You will need to keep track of the current `UITextField` state and set that state whenever the `ItemModel` is created. Also note in this example that whenever the text changes, we call `setItems()` on the `CollectionView` which will re-create our `ItemModels`.

For editable fields it's important to make sure you have the same `dataID` between edits and to rely on the content's `Equatable` conformance to inform Epoxy when it needs to reset the content. You should also make sure in your component that houses a `UITextField` that you check if the text field's text needs updating before setting the new value. Something like:

```
public func setContent(_ content: Content, animated: Bool) {
  if textField.text != content.text {
    textField.text = content.text
  }
}
```

To answer the question in the title of this post, Epoxy generates a `reuseID` based on [the view class and the hash of a view's `Style` instance](https://github.com/airbnb/epoxy-ios/wiki/EpoxyCollectionView#a-note-about-reuse-and-styles). So, if you have multiple components with the same style, the UICollectionView will reuse those cells as you scroll.

**Comment by rakuyoMo** [comment]
@thedrick Thank you for your answer. At first I also wrote my own code according to the code in `TextFieldViewController`. That is, I called the `setItem` method again to refresh the list as soon as there is input

But then I found out that on some systems (iOS15 and below), when users enter Chinese, there are some bugs in doing so: specifically, when they enter part of the Chinese, the keyboard will become English only, and they can't continue to enter Chinese.

I'm going to provide a demo that reproduces this problem in four to five hours
